test/functional/api/ui_spec.lua

```diff
@@ -0,0 +1,39 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local eq = helpers.eq
+local eval = helpers.eval
+local request = helpers.request
+
+describe('nvim_ui_attach()', function()
+  before_each(function()
+    clear()
+  end)
+  it('handles very large width/height #2180', function()
+    local screen = Screen.new(999, 999)
+    screen:attach()
+    eq(999, eval('&lines'))
+    eq(999, eval('&columns'))
+  end)
+  it('invalid option returns error', function()
+    local screen = Screen.new()
+    local status, rv = pcall(function() screen:attach({foo={'foo'}}) end)
```

Comment:
why not use `meths.ui_attach(80, 24, {foo='foo'})` directly? `Screen` doesn't do anything for us in this test.